src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -142,7 +156,8 @@ def _create_binary(self, binary_tgt, results_dir):
 
       # Add interpreter compatibility constraints to pex info. This will first check the targets for any
       # constraints, and if they do not have any will resort to the global constraints.
-      pex_builder.add_interpreter_constraints_from(constraint_tgts)
+      self._validate_interpreter_constraints(constraint_tgts)
```

Comment:
This is redundant. `interpreter = self.context.products.get_data(PythonInterpreter)` above already was priduced by a task that checked an even more stringent requirement - namely that all targets in the context with constraints have a local satisfying interpreter - the one we're building the pex with.